components/brave_wallet/resources/solana_provider.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function () {
+  if (!window.solana) {
+    return
+  }
+  const solanaWeb3 = require('@solana/web3.js')
+  window.solana.createPublickey = function createPublickey(base58Str) {
+    const result = new Object()
+    result.publicKey = new solanaWeb3.PublicKey(base58Str)
```

Comment:
since this is emitting a publicKey from the API call, do we need to be concerned about this API being used as a fingerprinting technique where the public key gets treated like a common public identifier to link a user across various websites?

Reply:
In order to do that, a website has to find a place to store a valid base58 encoded public key and share it among different parties, if the website wants to get it from solana wallet, that requires user permission to get it. At this point, it is no difference than calling `window.solana.publicKey`

Reply:
Ok, thanks for clarifying about this. I'm satisfied with this as is then. 

I'd missed that part which would make this far less likely to occur. Additionally, I think that's an acceptable tradeoff since the user consent is being requested when coming from the wallet. Anything above and beyond these techniques will require opening a can of worms I'd think which is well beyond the scope of this PR.